cmd/common-methods.go

```diff
@@ -32,6 +34,30 @@ import (
 	"github.com/minio/minio-go/v6/pkg/encrypt"
 )
 
+// decode if the key is encoded key and returns the key
+func getDecodedKey(sseKeys string) (key string, err *probe.Error) {
+	keyString := ""
+	for i, sse := range strings.Split(sseKeys, ",") {
+		if i > 0 {
+			keyString = keyString + ","
+		}
+		encryptString := strings.SplitN(sse, "=", 2)
+		pre := encryptString[0]
+		secretValue := encryptString[1]
+		decodedString, e := base64.StdEncoding.DecodeString(secretValue)
```

Comment:
It is better to have this part in a function. Other than that, the first part should be to check if the length of `secretValue` is `32`, then it is `plain text` and use it as is. Only otherwise do you proceed to `decode` and if the decode fails, you return error. Else check if the decoded length is `32`, then use that, else return `error`

Now, what will happen is that even if you encode a string of length `10` and pass it in command line, this code is going to use it as plain text as the decoded string length will be 10 and this code will assume that it is a plain text secret key.

Reply:
Added a function `func parseKey(sseKeys string) (sse string, err *probe.Error)` to achieve this functionality.